src/createAsyncThunk.ts

```diff
@@ -103,6 +103,64 @@ type GetRejectValue<ThunkApiConfig> = ThunkApiConfig extends {
   ? RejectValue
   : unknown
 
+type AsyncThunkReturnValue<ThunkArg, FulfilledValue, RejectedValue> =
+  | PayloadAction<FulfilledValue, string, { arg: ThunkArg; requestId: string }>
+  | PayloadAction<
+      undefined | RejectedValue,
+      string,
+      { arg: ThunkArg; requestId: string; aborted: boolean },
+      SerializedError
+    >
+/**
+ * A ThunkAction created by `createAsyncThunk`.
+ * Dispatching it returns a Promise for either a
+ * fulfilled or rejected action.
+ * Also, the returned value contains a `abort()` method
+ * that allows the asyncAction to be cancelled from the outside.
+ *
+ * @public
+ */
+export type AsyncThunkAction<
+  Returned,
+  ThunkArg,
+  ThunkApiConfig extends AsyncThunkConfig
+> = (
+  dispatch: GetDispatch<ThunkApiConfig>,
+  getState: () => GetState<ThunkApiConfig>,
+  extra: GetExtra<ThunkApiConfig>
+) => Promise<
+  AsyncThunkReturnValue<ThunkArg, Returned, GetRejectValue<ThunkApiConfig>>
+> & {
+  abort(reason?: string): void
+}
+
+type AsyncThunkActionCreator<
```

Comment:
I sure hope you know what you're doing here, because my eyes are glazing over

Reply:
I added more tests than I added code, guess we'll be fine :)